### PR TITLE
Getting our data right

### PR DESCRIPTION
**Pull request by lauralindal**
So, this is kind of a weird ticket. But it solves the question of when and where we want to get which data.
These are the steps 

1) Load seed Data for User from JSON file (this step is only needed for the prototype) on Sign Up and give it to hoodie
2) Load Hoodies data on Sign In and put it into state
3) If page is refreshed, get hoodie data and put it into state
4) On toggle, give data to hoodie.

Solves #58 and #39 too :)

### CODE REVIEW

**Comment by lauralindal** [comment]
Dear @schultyy, would you have a look? :)

File: src/Home.js [hunk 1]
```diff
@@ -113,39 +141,39 @@ class Home extends React.Component {
       }
     }
     for (var i = 0; i < selectedModuleIds.length; i++) {
-        for (var j=0; j < courseInfo.length; j++){
-          if (selectedModuleIds[i]===courseInfo[j].related_module_id)
-          selectedCourseData.push(courseInfo[j]);
-        }
+      for (var j=0; j < courseInfo.length; j++){
+        if (selectedModuleIds[i]===courseInfo[j].related_module_id)
```
Comment (schultyy) [inline src/Home.js]: I'd advise to not omit braces for if statements, for loops, etc.

The reason is the following:
It is easy to overread that the line after the if clause belongs to it, so it is easy to assume that the newly added line still belongs to the if case even though it doesn't.

File: src/Home.js [hunk 2]
```diff
@@ -113,39 +141,39 @@ class Home extends React.Component {
...
 
   retrieveSelectedModuleTitel(){
-     var userModules = this.state.userModules;
-     var modules = moduleplan.degree_course.modules;
-     var selectedModuleIds= [];
-     var selectedModuleTitels= [];
-     for (var i = 0; i < userModules.length; i++) {
+    var userModules = this.state.userModules;
+    var modules = moduleplan.degree_course.modules;
+    var selectedModuleIds= [];
```
Comment (schultyy) [inline src/Home.js]: There's a space missing around the `=`.

```
var selectedModuleIds = [];
```

File: src/Home.js [hunk 3]
```diff
@@ -113,39 +141,39 @@ class Home extends React.Component {
...
+    var selectedModuleTitels= [];
```
Comment (schultyy) [inline src/Home.js]: Same as above.

File: src/Home.js [hunk 4]
```diff
@@ -113,39 +141,39 @@ class Home extends React.Component {
...
         selectedModuleIds.push(userModules[i].module_id);
       }
     }
     for (var i = 0; i < selectedModuleIds.length; i++) {
-        for (var j=0; j < modules.length; j++){
-          if (selectedModuleIds[i]===modules[j].id)
-          selectedModuleTitels.push(modules[j].title);
-        }
+      for (var j=0; j < modules.length; j++){
+        if (selectedModuleIds[i]===modules[j].id)
```
Comment (schultyy) [inline src/Home.js]: Same case as with the if statement above.

File: src/Home.js [hunk 5]
```diff
@@ -158,18 +186,24 @@ class Home extends React.Component {
...
-       userModules[i].selected= !userModules[i].selected;
-       if (userModules[i].selected) {
-         userModules[i].status = "selected";
-       } else {
-         userModules[i].status = this.state.originalStatus[i];
-       }
+        if(userModules[i].status==="completed"){
+          return;
+        }
+        userModules[i].selected= !userModules[i].selected;
```
Comment (schultyy) [inline src/Home.js]: There's a space missing left of the assignment. Same as above.

File: src/user.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import users from './users.json';
+
+export function seedUserData(){
+  var userModules = users.students[0].tracked_modules;
+  return hoodie.store.add({
+    "userModules": userModules
+  })
+  .then(() => {
+    console.log('your initial data has been given to hoodie');
+    return userModules;
```
Comment (schultyy) [inline src/user.js:10]: `userModules` is undefined here. If you would like to return some data here, it should look like this: `return Promise.resolve(userModules);`.

File: src/Home.js [hunk 7]
```diff
@@ -13,25 +14,52 @@ class Home extends React.Component {
...
       hoodie.account.signUp({username: email, password: password});
+      seedUserData()
+      .then(function (userModules) {
+        console.log('your initial data has been saved in state');
+        self.setState({
+          userModules: userModules
+        });
+      });
       console.log('🐳', error)
     })
```
Comment (schultyy) [inline src/Home.js:61]: Semicolon is missing here.

File: src/Home.js [hunk 8]
```diff
@@ -113,39 +141,39 @@ class Home extends React.Component {
...
-        }
+      for (var j=0; j < courseInfo.length; j++){
+        if (selectedModuleIds[i]===courseInfo[j].related_module_id)
+        selectedCourseData.push(courseInfo[j]);
       }
+    }
     return selectedCourseData;
   };
 
   retrieveSelectedModuleTitel(){
```
Comment (schultyy) [inline src/Home.js]: Shouldn't this be `retrieveSelectedModuleTitle`?

File: src/Home.js [hunk 9]
```diff
@@ -113,39 +141,39 @@ class Home extends React.Component {
...
 
   combineSelectedTitelsAndData(){
     var courseInformation= this.retrieveSelectedCourseInfo();
     var modules = moduleplan.degree_course.modules;
-     for (var i = 0; i < modules.length; i++) {
-        for (var j=0; j < courseInformation.length; j++)
-          if (modules[i].id == courseInformation[j].related_module_id){
+    for (var i = 0; i < modules.length; i++) {
+      for (var j=0; j < courseInformation.length; j++)
+      if (modules[i].id == courseInformation[j].related_module_id){
```
Comment (schultyy) [inline src/Home.js]: Here you should use three equal signs for comparison to prevent automatic type coercion.

File: src/Home.js [hunk 10]
```diff
@@ -158,18 +186,24 @@ class Home extends React.Component {
...
-       if(userModules[i].status==="completed"){
-        return;
-       }
-       userModules[i].selected= !userModules[i].selected;
-       if (userModules[i].selected) {
-         userModules[i].status = "selected";
-       } else {
-         userModules[i].status = this.state.originalStatus[i];
-       }
+        if(userModules[i].status==="completed"){
```
Comment (schultyy) [inline src/Home.js]: Spaces are missing around `===`.

File: src/Home.js [hunk 11]
```diff
@@ -158,18 +186,24 @@ class Home extends React.Component {
     var data=null;
     for (var i = 0; i < userModules.length; i++) {
       if (userModules[i].module_id===moduleId){
```
Comment (schultyy) [inline src/Home.js]: Spaces are missing around `===`.

File: src/Home.js [hunk 12]
```diff
@@ -158,18 +186,24 @@ class Home extends React.Component {
...
       }
     }
-    this.setState({userModules:userModules});
+    this.setState({userModules: userModules});
+    hoodie.store.add({
+      "userModules": userModules
+    })
+    .then(() => {
+      console.log('hoodie now has your new data');
+    })
```
Comment (schultyy) [inline src/Home.js:207]: A semicolon is missing here.

**Review by schultyy — CHANGES_REQUESTED** [review]
Requesting minor changes, apart from that looks good to me 🙆

File: src/user.js (hunk 6 above)
Comment (lauralindal) [reply src/user.js:10]: I believe in this case, it's not undefined... We're creating the Variable userModules already before, so it should exist independently from the Promise?

Comment (schultyy) [reply src/user.js:10]: Yes, you're right I didn't read carefully here. My mistake.

**Review by schultyy — APPROVED** [review]

File: src/user.js (hunk 6 above)
Comment (lauralindal) [reply src/user.js:10]: No problem :) (But yayyyyy for me being right for once :D)

**Pull request merged**